pkg/build/gobuild.go

```diff
@@ -75,38 +81,80 @@ func (gbo *gobuildOpener) Open() (Interface, error) {
 		build:                gbo.build,
 		disableOptimizations: gbo.disableOptimizations,
 		mod:                  gbo.mod,
+		buildContext:         gbo.buildContext,
 	}, nil
 }
 
 // https://golang.org/pkg/cmd/go/internal/modinfo/#ModulePublic
+type modules struct {
+	main *modInfo
+	deps map[string]*modInfo
+}
+
 type modInfo struct {
 	Path string
 	Dir  string
+	Main bool
 }
 
 // moduleInfo returns the module path and module root directory for a project
 // using go modules, otherwise returns nil.
 //
 // Related: https://github.com/golang/go/issues/26504
-func moduleInfo() *modInfo {
-	output, err := exec.Command("go", "list", "-mod=readonly", "-m", "-json").Output()
+func moduleInfo() (*modules, error) {
+	modules := modules{
+		deps: make(map[string]*modInfo),
+	}
+
+	// TODO we read all the output as a single byte array - it may
+	// be possible & more efficient to stream it
+	output, err := exec.Command("go", "list", "-mod=readonly", "-json", "-m", "all").Output()
 	if err != nil {
-		return nil
+		return nil, nil
 	}
-	var info modInfo
-	if err := json.Unmarshal(output, &info); err != nil {
-		return nil
+
+	dec := json.NewDecoder(bytes.NewReader(output))
+
+	for {
+		var info modInfo
+
+		err := dec.Decode(&info)
+		if err == io.EOF {
+			// all done
+			break
+		}
+
+		modules.deps[info.Path] = &info
+
+		if info.Main {
+			modules.main = &info
```

Comment:
couldn't we hit N of these?  Which one wins?

Reply:
I believe there's only one

Reply:
main != `package main` but specifically "main module" = root module

Reply:
that's not confusing *at all* 🙄